### ISSUE TITLE
Wrong prefer-destructuring violation when variable already declared

### ISSUE THREAD

**Issue by berniegp** [eslint/eslint#11394]
**Tell us about your environment**

* **ESLint Version:** 5.12
* **Node Version:** v8.12.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** Default (I think, see below)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    root: true,
    extends: [
      'airbnb-base',
    ],
    plugins: [
      'vue'
    ],
    rules: {
        // Finding good names is hard so allow reuse
        'no-param-reassign': 0,

        // Finding good names is hard so allow reuse
        'no-shadow': 0,
    },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function myFunc(a, b) {
  // The following line violates "prefer-destructuring" even though it's not possible to use it
  // since a is already defined.
  a = otherFunction(b).a;
}
```

```bash
npx eslint index.js
```

**What did you expect to happen?**
No violation of "prefer-destructuring" since it's not possible to use destructuring here.

**What actually happened? Please include the actual, raw output from ESLint.**
./index.js
  1:10  error  'myFunc' is defined but never used  no-unused-vars
  2:3   error  Use object destructuring            prefer-destructuring
  2:7   error  'otherFunction' is not defined      no-undef

**Comment by platinumazure** [comment]
It's possible to destructure in an assignment statement-- you just need to surround the expression in parentheses:

```js
({a} = otherFunction(b));
```

Am I missing anything?

**Comment by berniegp** [comment]
Ah I didn't try with the parentheses. That is valid syntax indeed.

After more carefully reading the doc, I notice it's even mentioned there:

```
let foo;
({ foo } = object);
```
